### ISSUE TITLE
Server endpoint not retrieving payload from Shaka license request

### ISSUE THREAD

**Issue by davidnjohansson** [google/shaka-player#2352]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
YES

**What version of Shaka Player are you using?**
2.5.8

**Please ask your question**
I'm trying to setup encryption/decryption with Clearkey. I have created a Node Express POST endpoint to handle the license request. The request hits the endpoint but the request body which should contain the key ids is empty when it lands on the server. If I inspect the request in the Chrome network tab I can actually see the payload with the key ids which made me think the problem was server side, but when I do a JSON POST request with data in the body from e.g. Postman to the same endpoint it works just fine.

![request_payload](https://user-images.githubusercontent.com/24542784/73004832-423a5100-3e08-11ea-976b-e9bc7e7981e0.png)

So I'm not sure where things go wrong here.

Anyone have any clues?

**Comment by joeyparrish** [comment]
Can you share your express routes?  Can you also share the request headers?

**Comment by davidnjohansson** [comment]
I'm using NestJS which uses Express features. I have only one route defined, the one where I request the license, `http://localhost:3000/request-license`

Any POST from Postman outputs the JSON in the request body, it's when the request is sent from Shaka the request body is empty.

```
import { Controller, Post, Req } from '@nestjs/common';
import { Request } from 'express';

@Controller()
export class AppController {
  constructor() {}

  @Post('/request-license')
  requestLicense(@Req() request: Request) {
    console.log(request.body); // If sent from Shaka, outputs an empty JSON like so -> {}
  }
}
```

```
General:

Request URL: http://localhost:3000/request-license
Request Method: POST
Status Code: 201 Created
Remote Address: [::1]:3000
Referrer Policy: no-referrer-when-downgrade
```

```
Request Headers:

Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9,sv;q=0.8
Connection: keep-alive
Content-Length: 104
Host: localhost:3000
Origin: http://localhost:4200
Referer: http://localhost:4200/
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-site
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36
```

```
Request Payload:

{"kids":["88XgNh5mVLKPgEnHeLI5Rg","bXbyXLF_Xha46u9rv1gtjg","q7onHovPVSu9LoakNKml2Q"],"type":"temporary"}
```

**Comment by joeyparrish** [comment]
How does this compare with the request headers from Postman?

**Comment by davidnjohansson** [comment]
Here are the Postman request headers. 

All auto-generated except `Content-Type` which is set to `application/json`. If I remove the Content-Type header in Postman I get the same result as I get with the Shaka POST request, i.e. and empty body on the server.

Could it just be that we need to set `Content-Type` to `application/json` in the Shaka POST request?

![postman_request_headers](https://user-images.githubusercontent.com/24542784/73021371-bc2e0280-3e27-11ea-865a-6e43bb5b3d7d.png)

**Comment by joeyparrish** [comment]
Yes, I see two solutions:

1. You can register a request filter in Shaka Player to set the `Content-Type` header in the outgoing license request.  See example below.
2. You can tell NestJS **not** to expect JSON, but rather to let you have the request body.  (Not sure what format NestJS gives you this.)  You can then convert it to a string (if it's not already) and run it through `JSON.parse` yourself.

Here's what the player-based solution looks like:
```js
player.getNetworkingEngine.registerRequestFilter((type, request) => {
  if (type == shaka.net.NetworkingEngine.RequestType.LICENSE) {
    request.headers['Content-Type'] = 'application/json';
  }
});
```

Does this help?

**Comment by davidnjohansson** [comment]
I went for the player-based solution and it works now, thank you very much for the support 👍

PS you forgot the parentheses after `player.getNetworkingEngine`.. just in case someone else is interested in that snippet.. &#128521;

**Comment by joeyparrish** [comment]
Thanks for catching that.  I've edited the code snippet with the fix.
